test/unit/controllers/declaration/ExporterEoriNumberControllerSpec.scala

```diff
@@ -0,0 +1,209 @@
+
+        theResponseForm.value.value.eori mustBe Some(Eori(eori))
+        theResponseForm.value.value.hasEori mustBe hasEori
+      }
+
+      "display page method is invoked and cache contains no Exporter data" in {
+
+        withNewCaching(aDeclarationAfter(request.cacheModel))
+
+        val result = controller.displayPage(Mode.Normal)(getRequest())
+
+        status(result) mustBe OK
+        checkViewInteractions()
+
+        theResponseForm.value mustBe None
+      }
+    }
+
+    "should return a 400 (BAD_REQUEST)" when {
+      "EORI is incorrect" in {
+
+        withNewCaching(request.cacheModel)
+
+        val incorrectForm = Json.toJson(ExporterEoriNumber(eori = Some(Eori("!@#$")), hasEori = YesNoAnswers.yes))
+
+        val result = controller.submit(Mode.Normal)(postRequest(incorrectForm))
+
+        status(result) mustBe BAD_REQUEST
+        checkViewInteractions()
+      }
+
+      "EORI is not provided but trader selected that it has an EORI" in {
+
+        withNewCaching(request.cacheModel)
+
+        val incorrectForm = Json.toJson(ExporterEoriNumber(eori = None, hasEori = YesNoAnswers.yes))
+
+        val result = controller.submit(Mode.Normal)(postRequest(incorrectForm))
+
+        status(result) mustBe BAD_REQUEST
+        checkViewInteractions()
+      }
+
+      "no choice is selected and no cached ExporterDetails exist" in {
+
+        withNewCaching(request.cacheModel)
+
+        val correctForm = Json.toJson(ExporterEoriNumber(eori = None, hasEori = ""))
+
+        val result = controller.submit(Mode.Normal)(postRequest(correctForm))
+
+        status(result) mustBe BAD_REQUEST
+        checkViewInteractions()
+      }
+    }
+
+    "should return a 303 (SEE_OTHER)" when {
+      "'No' is selected" in {
+
+        withNewCaching(request.cacheModel)
+
+        val correctForm = Json.toJson(ExporterEoriNumber(eori = None, YesNoAnswers.no))
+
+        val result = controller.submit(Mode.Normal)(postRequest(correctForm))
+
+        await(result) mustBe aRedirectToTheNextPage
+        thePageNavigatedTo mustBe controllers.declaration.routes.ExporterDetailsController.displayPage()
+        checkViewInteractions(0)
+        theCacheModelUpdated.parties.exporterDetails must be(Some(ExporterDetails(EntityDetails(None, None))))
+      }
+
+      "'Yes' is selected" in {
+
+        withNewCaching(request.cacheModel)
+
+        val eoriInput = Some(Eori("GB123456789000"))
+        val correctForm = Json.toJson(ExporterEoriNumber(eori = eoriInput, YesNoAnswers.yes))
+
+        val result = controller.submit(Mode.Normal)(postRequest(correctForm))
+
+        await(result) mustBe aRedirectToTheNextPage
+        if (request.declarationType == DeclarationType.CLEARANCE) {
+          thePageNavigatedTo mustBe controllers.declaration.routes.IsExsController.displayPage()
+        } else {
+          thePageNavigatedTo mustBe controllers.declaration.routes.RepresentativeAgentController.displayPage()
+        }
```

Comment:
I see what you did there ;-)
I'd rather stay with the pattern we have, so to extract this test case and use `onJourney(CLEARANCE)`. I'm not saying it's wrong, but I see consistency a big issue in our project.

Reply:
This has been addressed. Please review.